app/uk/gov/hmrc/customs/notification/repo/LockRepo.scala

```diff
@@ -16,22 +16,73 @@
 
 package uk.gov.hmrc.customs.notification.repo
 
-import org.joda.time.DateTime
-import uk.gov.hmrc.customs.notification.domain.{ClientNotification, ClientSubscriptionId}
 
-import scala.concurrent.Future
-import scala.concurrent.duration.Duration
+import org.joda.time.Duration
+import reactivemongo.api.DB
+import uk.gov.hmrc.customs.notification.domain.ClientSubscriptionId
+import uk.gov.hmrc.lock.{ExclusiveTimePeriodLock, LockRepository}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+
+class LockOwnerId(val lockOwnerId: String) extends AnyVal
 
-/**
-  * Created by dev on 25/06/2018.
-  */
 trait LockRepo {
 
+  val db: () => DB
+  val repo = new LockRepository()(db)
 
-  def lock(csid: ClientSubscriptionId, duration: Duration): Future[Boolean]
+  /*
+    Calling lock will try to renew a lock but acquire a new lock if it doesn't exist
+   */
+  def lock(csId: ClientSubscriptionId,lockOwnerId : LockOwnerId, duration: Duration): Future[Boolean] = {
+    val lock: ExclusiveTimePeriodLock = new NotificationExclusiveTimePeriodLock(csId,lockOwnerId, duration, db, repo)
+    val eventualMaybeBoolean: Future[Option[Boolean]] = lock.tryToAcquireOrRenewLock(Future.successful(true))
+    val eventualBoolean: Future[Boolean] = eventualMaybeBoolean.map {
+      case Some(true) => true
+      case _ => false
+    }
+    eventualBoolean
 
-  def release(csid: ClientSubscriptionId): Future[Unit]
+  }
 
+  /*
+   Calling release lock will call directly to the repository code
+  */
+  def release(csId: ClientSubscriptionId, lockOwnerId: LockOwnerId): Future[Unit] = {
+    val lock: NotificationExclusiveTimePeriodLock = new NotificationExclusiveTimePeriodLock(csId, lockOwnerId,  Duration.ZERO, db, repo)
```

Comment:
open question do we we need logging at debug level for all lock operations?

Reply:
?? We did wonder about logging but thought if we added it now when we dont have all interactions we might have too much. We can add it now but might need discussion